packages/site/cms-site/src/sitePreview/SitePreviewUtils.ts

```diff
@@ -4,6 +4,7 @@ import { jwtVerify, SignJWT } from "jose";
 import { cookies, draftMode } from "next/headers";
 import { redirect } from "next/navigation";
 import { type NextRequest } from "next/server";
+import path from "path";
```

Comment:
Optional: I wouldn't use path (Node API) for this, but `URL`. Next tends to lean towards using more web APIs and less Node APIs

Reply:
I don't see how URL would help me with building a path? I still have to build the path to override URL.pathname.

My main motivation for `path.join` was that it helps me with avoiding double-slashes because it transforms something like this `//en///abc/` into this `/en/abc/`. 

I could alternatively do it manually like `/${basePath}/${path}`. But then I have to handle double slashes myself which is a pain.

Reply:
Maybe I haven't thought this through 😅 Anyway, it was only an suggestion, so we can leave it like this.

Reply:
you are right, path is for filesystem paths (with backslashes on windows etc), not for Urls

Reply:
> you are right, path is for filesystem paths (with backslashes on windows etc), not for Urls

I'm aware, but since we don't support Windows, it doesn't matter in practice

Anyway, I now implemented it myself: ac9fb18b2b44fbfcff2762cd7e0a49776dbbeb3b